src/js/services/auth.js

```diff
@@ -33,15 +30,62 @@ export default App.extend({
   getCurrentOrg() {
     return this.currentOrg;
   },
+  getTokenExpiration() {
+    return expires ? Date.parse(expires) : new Date();
+  },
+  isTokenExpired() {
+    return this.getTokenExpiration() < (new Date());
+  },
+  removeToken() {
+    token = null;
+    expires = null;
+    sessionStorage.removeItem('auth:token');
+    sessionStorage.removeItem('auth:expires');
+  },
   initBootstrap() {
-    const d = $.Deferred();
+    if (this.isTokenExpired()) {
+      this.removeToken();
+    }
+
+    if (token) {
+      const d = $.Deferred();
+      $.when(Radio.request('entities', 'fetch:clinicians:current')).then(currentUser => {
+        this.currentUser = currentUser;
+        this.currentOrg = this.currentUser.getOrganization();
+        d.resolve(currentUser);
+      });
+      return d.promise();
+    }
 
-    $.when(Radio.request('entities', 'fetch:clinicians:current')).then(currentUser => {
-      this.currentUser = currentUser;
-      this.currentOrg = this.currentUser.getOrganization();
-      d.resolve(currentUser);
+    // https://auth0.com/docs/libraries/auth0js/v9#webauth-authorize-
+    const auth = new auth0.WebAuth({
+      clientID: '2eg7mz2db0B31gctULMTcgax1vdvgPip',
+      domain: 'roundingwell-care-team.auth0.com',
+      responseType: 'id_token',
```

Comment:
So from the meeting I gathered that the `token` is what is needed for the refresh?..  I'm not sure what `responseMode` `fragment` does with this, but `responseType` can be space separated: `'id_token token'` to apparently somehow return both?  Either way, it doesn't seem like we need to worry about this for MVP, but it might be possible to get both?

Reply:
@paulfalgout we can get both, but access tokens (`token`) is the only thing that can be refreshed and it does not contain any user info. The ID token is generated once per login and cannot be refreshed.

Reply:
Seems to be an issue a lot of people are dealing with... most questions go unanswered, and the few that do... not great responses: https://community.auth0.com/t/anyone-refreshing-tokens-id-token-missing-in-refresh-token-response/7067/2 https://community.auth0.com/t/oauth-token-not-returning-id-token/8303

For another day I guess.

Reply:
Oh I should also mention: ID tokens last for 24 hours, so most likely users would only need to login once a day.

Reply:
makes me think maybe we should set the expire to 10 hours or something so that someone logging in at the end of the day is more likely to have to relogin each day instead of having a chance to hit that 24 hour mark the next day

Reply:
Well, I'm using `sessionStorage` so that if the user closes the tab or browser they will need to login again. I think what we have is fine until proven otherwise.